include/dmlc/registry.h

```diff
@@ -189,6 +189,11 @@ class FunctionRegEntryBase {
     info.type = type;
     info.type_info_str = info.type;
     info.description = description;
+    /*! \brief Check that this is not a duplicate argument */
+    DCHECK_EQ(std::find_if(arguments.begin(), arguments.end(),
```

Comment:
There were several places in mxnet where attributes were being added more than once.  This will catch the superfluous addition in debug mode.